src/oidc_clients/api/user_consent.ts

```diff
@@ -0,0 +1,26 @@
+// api for fetching user clients
+import { IAuthUser } from 'authentication/models/User';
+import { deleteR, getAllPages } from 'common/utils/api';
+import { IOidcClient } from 'oidc_clients/models/client';
+import { IUserConsent } from '../models/user_consent';
+import { getClient } from './clients';
+
+const API_BASE = '/api/v1/oidc/';
+
+type ConsentResponseType = { client: number | IOidcClient } & Omit<IUserConsent, 'client'>;
+
+export const getUserConsent = async (user: IAuthUser) => {
+  const consents: IUserConsent[] = [];
+  const consentResponse = await getAllPages<ConsentResponseType>(API_BASE + 'consent/', undefined, { user });
+
+  for (const consent of consentResponse) {
+    const client = await getClient(consent.client as number);
+    consent.client = client;
+    consents.push(consent as IUserConsent);
+  }
```

Comment:
This would still be a synchronous loop, would it not?
We could speed it up a bit by mapping a `Promise.all`?